From 84a39ed832e749d7a7161dc9a976e5d67c47bfd8 Mon Sep 17 00:00:00 2001
From: Augusto Righetto <aurighet@microsoft.com>
Date: Thu, 2 Jul 2020 15:06:44 -0700
Subject: [PATCH] Disabling switch without case warning for aec3

---
 modules/audio_processing/aec3/fft_data.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/audio_processing/aec3/fft_data.h b/modules/audio_processing/aec3/fft_data.h
index 5e5adb62de..7707249704 100644
--- a/modules/audio_processing/aec3/fft_data.h
+++ b/modules/audio_processing/aec3/fft_data.h
@@ -44,6 +44,8 @@ struct FftData {
   void Spectrum(Aec3Optimization optimization,
                 rtc::ArrayView<float> power_spectrum) const {
     RTC_DCHECK_EQ(kFftLengthBy2Plus1, power_spectrum.size());
+#pragma warning(push)
+#pragma warning(disable:4065)
     switch (optimization) {
 #if defined(WEBRTC_ARCH_X86_FAMILY)
       case Aec3Optimization::kSse2: {
@@ -65,6 +67,7 @@ struct FftData {
         std::transform(re.begin(), re.end(), im.begin(), power_spectrum.begin(),
                        [](float a, float b) { return a * a + b * b; });
     }
+#pragma warning(pop)
   }
 
   // Copy the data from an interleaved array.
-- 
2.24.1.windows.2

